### PR TITLE
CVSL-2177: Add resident endpoint.

### PR DESCRIPTION
**Pull request by justin-gardiner**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/os/OsPlacesApiDPA.kt [hunk 1]
```diff
@@ -48,3 +49,24 @@ data class OsPlacesApiDPA(
   @JsonProperty("LAST_UPDATE_DATE")
   val lastUpdateDate: LocalDate,
 )
+
+fun OsPlacesApiDPA.getAddressFirstLine(): String {
```
Comment (justin-gardiner) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/os/OsPlacesApiDPA.kt:53]: just moved these extension functions.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/model/ResidentSummary.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  @Schema(description = "The resident's surname", example = "Jones")
+  val surname: String,
+
+  @Schema(description = "The resident's phone number", example = "07634183674")
+  val phoneNumber: String? = null,
+
+  @Schema(description = "The resident's relation to the offender", example = "Mother")
+  val relation: String,
+
+  @Schema(description = "The resident's date of birth", example = "Mother")
```
Comment (louisenorris) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/model/ResidentSummary.kt:24]: **niggle:** tiny thing but think the example here should be a date

**Review by louisenorris — DISMISSED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/model/ResidentSummary.kt (hunk 2 above)
Comment (justin-gardiner) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/model/ResidentSummary.kt:24]: ahh yes good spot, I thought I fixed that

Comment (justin-gardiner) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/model/ResidentSummary.kt]: fixed

**Review by louisenorris — APPROVED** [review]
LGTM

**Pull request merged**